src/astro/databases/snowflake.py

```diff
@@ -75,3 +77,46 @@ def load_pandas_dataframe_to_table(
             chunk_size=chunk_size,
             quote_identifiers=False,
         )
+
+    def add_templates_to_context(self, parameters: Dict, context: Dict) -> Dict:
```

Comment:
@tatiana the alternative is that the operator now has to carry the complexity of each database's specific templating logic. This keeps to the very clean interface where someone can implement a new database only by creating a new database class (otherwise they would need to go into the operators to add each database and we're back where we started).

FWIW the database class is still agnostic to airflow stuff as in this case the context is just a function parameter, instead of something that the database class has to hold on to.